brainlit/algorithms/generate_fragments/spline_fxns.py

```diff
@@ -0,0 +1,296 @@
+import operator
+from scipy.interpolate import BSpline, splev
+import numpy as np
+from typing import Iterable, Union, Tuple
+from brainlit.utils.util import (
+    check_type,
+    check_size,
+    check_precomputed,
+    check_iterable_type,
+    check_iterable_nonnegative,
+)
+
+
+def speed(
+    x: np.ndarray,
+    t: np.ndarray,
+    c: np.ndarray,
+    k: np.integer,
+    aux_outputs: bool = False,
+) -> np.ndarray:
+    """Compute the speed of a B-Spline.
+
+    The speed is the norm of the first derivative of the B-Spline
+
+    Arguments:
+        x: A 1xL array of parameter values where to evaluate the curve
+        t: A 1xm array representing the knots of the B-spline
+        c: A dxn array representing the control points of the B-spline
```

Comment:
I'm used to referring to this as "coefficients," is there a reason you call it control points?

Reply:
I found a couple of sources that refer to them as _control points_

- https://web.mit.edu/hyperbook/Patrikalakis-Maekawa-Cho/node17.html
- https://mathworld.wolfram.com/B-Spline.html
- https://en.wikipedia.org/wiki/Trace_(linear_algebra)

But I agree that Kunoth calls them _coefficients_. We can probably keep "coefficients/control points"?